ph_rando/patcher/_util.py

```diff
@@ -301,6 +301,7 @@ def _patch_shop_items(areas: list[Area], input_rom: rom.NintendoDSRom) -> None:
 
     # Load arm9.bin and overlay table
     arm9_executable = bytearray(code.MainCodeFile(input_rom.arm9, 0x02000000).save(compress=False))
+    arm9_executable_o = arm9_executable
```

Comment:
I don't think this is sufficient, as you're just copying the reference to the bytearray, not creating a new one.

A simpler example of this that illustrates the issue:
```python
>>> arm9_executable = bytearray([1,2,3,4])
>>> arm9_executable_o = arm9_executable
>>> arm9_executable[0] = 5
>>> print(arm9_executable)
bytearray(b'\x05\x02\x03\x04')
>>> print(arm9_executable_o)
bytearray(b'\x05\x02\x03\x04')
```

You can fix this by using the `copy` module
```python
>>> from copy import copy
>>> arm9_executable = bytearray([1,2,3,4])
>>> arm9_executable_o = copy(arm9_executable)
>>> arm9_executable[0] = 5
>>> print(arm9_executable)
bytearray(b'\x05\x02\x03\x04')
>>> print(arm9_executable_o)
bytearray(b'\x01\x02\x03\x04')
```